src/pyodide-py/pyodide/console.py

```diff
@@ -1,8 +1,40 @@
-from typing import Optional, Callable
+from typing import Optional, Callable, Any
 import code
 import io
 import sys
 import platform
+from contextlib import contextmanager
+import builtins
+
+# this import can fail when we are outside a browser (e.g. for tests)
+try:
+    import js
+
+    dummy_promise = js.Promise.resolve()
+    load_packages_from_imports = js.pyodide.loadPackagesFromImports
+
+except ImportError:
+
+    class FakePromise:
```

Comment:
This doesn't need to be part of Public API? If so let's rename it,
```suggestion
    class _FakePromise:
```

Reply:
You're right. I'll make it private.